test/parallel/test-buffer-alloc.js

```diff
@@ -0,0 +1,1466 @@
+  assert.equal(c.compare(d), 1);
+  assert.equal(d.compare(b), 1);
+  assert.equal(b.compare(d), -1);
+  assert.equal(b.compare(b), 0);
+
+  assert.equal(Buffer.compare(b, c), -1);
+  assert.equal(Buffer.compare(c, d), 1);
+  assert.equal(Buffer.compare(d, b), 1);
+  assert.equal(Buffer.compare(b, d), -1);
+  assert.equal(Buffer.compare(c, c), 0);
+
+  assert.equal(Buffer.compare(Buffer.alloc(0),
+                              Buffer.alloc(0)), 0);
+  assert.equal(Buffer.compare(Buffer.alloc(0),
+                              Buffer.allocUnsafe(1)), -1);
+  assert.equal(Buffer.compare(Buffer.allocUnsafe(1),
+                              Buffer.alloc(0)), 1);
+}
+
+assert.throws(function() {
+  var b = Buffer.allocUnsafe(1);
+  Buffer.compare(b, 'abc');
+});
+
+assert.throws(function() {
+  var b = Buffer.allocUnsafe(1);
+  Buffer.compare('abc', b);
+});
+
+assert.throws(function() {
+  var b = Buffer.allocUnsafe(1);
+  b.compare('abc');
+});
+
+// Test Equals
+{
+  const b = Buffer.allocUnsafe(5).fill('abcdf');
+  const c = Buffer.allocUnsafe(5).fill('abcdf');
+  const d = Buffer.allocUnsafe(5).fill('abcde');
+  const e = Buffer.allocUnsafe(6).fill('abcdef');
+
+  assert.ok(b.equals(c));
+  assert.ok(!c.equals(d));
+  assert.ok(!d.equals(e));
+  assert.ok(d.equals(d));
+}
+
+assert.throws(function() {
+  var b = Buffer.allocUnsafe(1);
+  b.equals('abc');
+});
+
+// Regression test for https://github.com/nodejs/node/issues/649.
+assert.throws(
+  function() { Buffer.allocUnsafe(1422561062959).toString('utf8'); });
+
+var ps = Buffer.poolSize;
+Buffer.poolSize = 0;
+assert.equal(Buffer.allocUnsafe(1).parent, undefined);
+Buffer.poolSize = ps;
+
+// Test Buffer.copy() segfault
+assert.throws(function() {
+  Buffer.allocUnsafe(10).copy();
+});
+
+// Test prototype getters don't throw
+assert.equal(Buffer.prototype.parent, undefined);
+assert.equal(Buffer.prototype.offset, undefined);
+assert.equal(SlowBuffer.prototype.parent, undefined);
+assert.equal(SlowBuffer.prototype.offset, undefined);
+
+// Must be zero-filled.
+// This can be a bit unreliable because the allocated memory might
+// actually already be filled with zeroes... the test is iterated
+// multiple times to make it more reliable
+{
+  for (let n = 0; n < 50; n++) {
+    const buf = Buffer.alloc(10);
+    for (let i = 0; i < buf.length; i++)
+      assert.equal(buf[i], 0);
+  }
+}
+
+// Must be filled with 1's
+{
+  const buf = Buffer.alloc(10, 1);
+  for (let i = 0; i < buf.length; i++)
+    assert.equal(buf[i], 1);
+}
+
+// Must be filled with 'ABC', decoded from hex
+{
+  const check = [65, 66, 67];
+  const buf = Buffer.alloc(9, '414243', 'hex');
+  for (let i = 0; i < buf.length; i++)
+    assert.equal(buf[i], check[i % 3]);
+  assert.equal(buf.toString(), 'ABCABCABC');
+}
+
+// Throws due to invalid arguments
+assert.throws(() => Buffer.alloc('foo'), TypeError);
+assert.throws(() => Buffer.allocUnsafe('foo'), TypeError);
+assert.throws(() => Buffer.from(1), TypeError);
```

Comment:
It was copied directly from test-buffer.js. It likely can and should be broken down but I'd prefer to do that in a separate PR